docs/proposals/self-service-notifications.md

```diff
@@ -0,0 +1,144 @@
+---
+title: ArgoCD self-service notification
+
+authors:
+- "@mayzhang2000"
+- "@zachaller"
+- "@crenshaw-dev"
+- "@leoluz"
+
+sponsors:
+- TBD
+
+reviewers:
+- TBD
+
+approvers:
+- TBD
+
+creation-date: 2023-08-22  
+last-updated: 2023-08-22
+---
+
+# Self Service Notification for ArgoCD
+
+## Summary
+This proposal is to enable application teams to have their own configurations of ArgoCD notifications aka self-service notification.
+Application team will be able to receive notifications from the default (using the configuration from the namespace of the notification controller) as well as their own configuration of notifications.
+
+## Motivation
+As of now the configuration for ArgoCD notification is centrally managed. Only ArgoCD admin can make notification configuration changes.
+
+When application teams use PagerDutyV2 for their notification service, every application team needs to create an integration key for each service in Pager Duty.
+ArgoCD admin needs to add the integration key to kubernete's secret `argocd-notifications-secret`,
+also needs to modify configmap `argocd-notifications-cm` to add the reference to the integration key stored in above secret under the list of `serviceKeys`
+
+When there are many application teams want to use PagerDutyV2 for their notification service, they all have to go to ArgoCD admin team. This does not scale.
+
+Teams also want to control the notification templates and triggers independently of what ArgoCD admin configured.
+
+We need to enable application team's the ability to configure their own notification configurations.
+
+## Proposal
+Deploy notification configuration resources (configmap, secret) in the same namespace where the ArgoCD application is at.
+ArgoCD applications are in any namespace using apps in any namespace feature.
+
+Enhance ArgoCD notification controller to support app in any namespace.
+Notification controller will know the namespaces to monitor by using `--application-namespaces` startup parameter. 
+It can also be conveniently set up and kept in sync by specifying the application.namespaces settings in the argocd-cmd-params-cm ConfigMap.
+
+Enhance ArgoCD notification controller to use notification-engine's function `NewControllerWithNamespaceSupport()`. 
+This function uses the application's `metadata.namespace` field to find the location of the configmap and secret. 
+This controller uses both the self-service configuration and the admin defined default configuration.
```

Comment:
Does this proposal account for a conflict between a template or service name in the default configuration and the self-service configuration? For example, both define a `service.pagerduty` and `template.app-sync-running`. Will the self-service one take precedence?

Reply:
It goes through one configuration at a time. If both defined the same service and trigger, both notifications are sent.